src/resources/platform_handlers/pytorch_libtorch/model.py

```diff
@@ -0,0 +1,194 @@
+# and converting Triton input/output types to numpy types.
+import triton_python_backend_utils as pb_utils
+
+
+def _get_model_filename(config):
+    model_filename = config["default_model_filename"]
+    if not model_filename:
+        model_filename = "model.py"
+    return model_filename
+
+
+def _get_model_path(config):
+    model_filename = _get_model_filename(config)
+    model_path = os.path.join(pb_utils.get_model_dir(), model_filename)
+    if not os.path.exists(model_path):
+        raise pb_utils.TritonModelException(f"No model found in " + model_path)
+    return model_path
+
+
+def _get_model_data_path(config):
+    model_data_filename = _get_model_filename(config) + ".pt"
+    model_data_path = os.path.join(pb_utils.get_model_dir(), model_data_filename)
+    if not os.path.exists(model_data_path):
+        return ""  # pickled data not provided
+    return model_data_path
+
+
+def _import_module_from_path(module_name, file_path):
+    spec = importlib.util.spec_from_file_location(module_name, file_path)
+    module = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(module)
+    return module
+
+
+class TritonPythonModel:
+    """Your Python model must use the same class name. Every Python model
+    that is created must have "TritonPythonModel" as the class name.
+    """
+
+    def initialize(self, args):
+        """`initialize` is called only once when the model is being loaded.
+        Implementing `initialize` function is optional. This function allows
+        the model to initialize any state associated with this model.
+
+        Parameters
+        ----------
+        args : dict
+          Both keys and values are strings. The dictionary keys and values are:
+          * model_config: A JSON string containing the model configuration
+          * model_instance_kind: A string containing model instance kind
+          * model_instance_device_id: A string containing model instance device ID
+          * model_repository: Model repository path
+          * model_version: Model version
+          * model_name: Model name
+        """
+        self._model_name = args["model_name"]
+        self._model_config = json.loads(args["model_config"])
+
+        self._logger = pb_utils.Logger
+        self._logger.log_info("Initializing model instance for " + self._model_name)
+
+        model_path = _get_model_path(self._model_config)
+        data_path = _get_model_data_path(self._model_config)
+
+        self._model_module = _import_module_from_path(self._model_name, model_path)
+        self._model_class = getattr(self._model_module, self._model_name)
+        self._raw_model = self._model_class()
+        if data_path != "":
+            self._raw_model.load_state_dict(torch.load(data_path))
+        else:
+            self._logger.log_info("Model weights not found for " + self._model_name)
+        self._raw_model.eval()
+        self._model = torch.compile(self._raw_model)
```

Comment:
I think this should be an option and shouldn't be enabled by default.

Reply:
This is a good point. I think it would be better to allow opt-out instead of opt-in, as long as proper checks for its support are in place (i.e. PyTorch >= 2.0).

The user is able to disable it in parameters: [Add ability to disable torch.compile](https://github.com/triton-inference-server/python_backend/pull/282/commits/9e83e246c6f20948b7e9dcee82b3f8b1a30aad83)